DOC: Fixed the doctsring for _set_axis_name (GH 22895)

bkjchoi72 opened the pull request:
Fixed the docstring of the function `_set_axis_name` in `pandas/core/generic.py`

Closes #22895

bkjchoi72 committed:
DOC: Fixed the doctsring for _set_axis_name (GH 22895)

pep8speaks commented:
Hello @bkjchoi72! Thanks for submitting the PR.

 - There are no PEP8 issues in the file [`pandas/core/generic.py`](https://github.com/bkjchoi72/pandas/blob/a5fd1c33adeeabc1c4c6ecfb68c18f5959398ba3/pandas/core/generic.py) !

bkjchoi72 committed:
Added a new line after the first triple quotes

bkjchoi72 committed:
Removed the new line after the first triple quotes

bkjchoi72 committed:
Merge branch 'master' into GH22895_fix_docstring_for_set_axis_name

gfyoung reviewed (approved):
Nice!

cc @datapythonista

gfyoung commented:
@bkjchoi72 : FYI, you don't have to close your PR (#22968) and open a new if there are issues with it.  You can keep making commits to the existing PR and fix them there.

bkjchoi72 committed:
Merge branch 'master' into GH22895_fix_docstring_for_set_axis_name

bkjchoi72 committed:
Merge branch 'master' into GH22895_fix_docstring_for_set_axis_name

bkjchoi72 committed:
Modified the description of parameters. Changed the example to animals.

bkjchoi72 commented on pandas/core/generic.py:
```diff
@@ -1194,31 +1194,35 @@ def rename_axis(self, mapper, axis=0, copy=True, inplace=False):
-            whether to modify `self` directly or return a copy
+            Labels(s) to set.
+        axis : int or str, default 0
+            The axis to set the label. The value 0 or 'index' specifies index,
```
Thanks for the information

bkjchoi72 commented on pandas/core/generic.py:
```diff
@@ -1194,31 +1194,35 @@ def rename_axis(self, mapper, axis=0, copy=True, inplace=False):
+            The axis to set the label. The value 0 or 'index' specifies index,
+            and the value 1 or 'columns' specifies columns.
+        inplace : bool, default False
+            Whether to modify `self` directly or return a copy.
```
There isn't great consistency for the description of  `inplace`. The most common and easy to understand I found is `If `True`, do operation inplace and return None.`. I'll go with that. Also I'll change `bool` to `boolean`.

bkjchoi72 commented on pandas/core/generic.py:
```diff
@@ -1194,31 +1194,35 @@ def rename_axis(self, mapper, axis=0, copy=True, inplace=False):
 
         Examples
         --------
+        >>> df = pd.DataFrame({"A": [1, 2, 3]})
```
Thanks for the suggestion. I'll make those changes.

bkjchoi72 committed:
Modified the example for DataFrame.rename_axis()

datapythonista commented on pandas/core/generic.py:
```diff
@@ -1154,29 +1154,32 @@ def rename_axis(self, mapper, axis=0, copy=True, inplace=False):
+        0          dog     4
+        1          cat     4
+        2       monkey     2
+        >>> df.rename_axis("id", axis="columns")
```
In this case, the concept `id` refers to `name` and `legs`, which does not make sense.

It could make more sense to have the name as the index, and then may be a column `num_legs` and a column `num_arms` and call the axis `limbs`.

bkjchoi72 replied:
Makes sense

datapythonista commented on pandas/core/generic.py:
```diff
@@ -1154,29 +1154,32 @@ def rename_axis(self, mapper, axis=0, copy=True, inplace=False):
-        0    1  4
-        1    2  5
-        2    3  6
+        >>> df = pd.DataFrame({"name": ["dog", "cat", "monkey"], "legs": [4, 4, 2]})
```
this is longer than 79 characters, isn't it?

datapythonista reviewed (commented):
looking good, just two small things

bkjchoi72 committed:
Changed the example so that it makes sense

datapythonista commented on pandas/core/generic.py:
```diff
@@ -1154,29 +1154,34 @@ def rename_axis(self, mapper, axis=0, copy=True, inplace=False):
-        1    2  5
-        2    3  6
+        >>> df = pd.DataFrame({"num_legs": [4, 4, 2],
+        ...                   "num_arms": [0, 0, 2]},
```
the indentation of this line looks wrong

datapythonista commented on pandas/core/generic.py:
```diff
@@ -1154,29 +1154,34 @@ def rename_axis(self, mapper, axis=0, copy=True, inplace=False):
-        2    3
-        dtype: int64
+        >>> s = pd.Series(["dog", "cat", "monkey"])
+        >>> s.rename_axis("animal")
```
can you show what's the name of the axis before renaming it

datapythonista replied:
Can you add this, it was originally requested by @jreback, and it makes sense to have. After this we can merge.

bkjchoi72 replied:
Doesn't line 1158 show the contents of `s` including the name of the axis which is not set at the time?

bkjchoi72 replied:
I think there is a bug in Git. If you go into `Files changed`, it shows you that line 1158 prints the value of `s`

datapythonista replied:
@jreback ?

datapythonista commented on pandas/core/generic.py:
```diff
@@ -1194,44 +1199,54 @@ def rename_axis(self, mapper, axis=0, copy=True, inplace=False):
     def _set_axis_name(self, name, axis=0, inplace=False):
         """
-        Alter the name or names of the axis.
+        Alter the label(s) of the axis.
```
This doesn't seem correct. Altering the labels would be changing the names of the columns, not the name of the axis itself.

And small thing, but alter doesn't necessarily need to be correct. If the axis doesn't have a name, it's setting it, but not altering. `Set` would be better I think.

datapythonista commented on pandas/core/generic.py:
```diff
@@ -1194,44 +1199,54 @@ def rename_axis(self, mapper, axis=0, copy=True, inplace=False):
-           0 or 'index' for the index; 1 or 'columns' for the columns
-        inplace : bool
-            whether to modify `self` directly or return a copy
+            Labels(s) to set.
```
Same as before, label is the names of each individual column, not the name of the axis

datapythonista commented on pandas/core/generic.py:
```diff
@@ -1194,44 +1199,54 @@ def rename_axis(self, mapper, axis=0, copy=True, inplace=False):
         Returns
         -------
-        renamed : same type as caller or None if inplace=True
+        Series, DataFrame, Panel, or None
```
Can you remove `Panel` here. While it's correct, Panel it's already deprecated and will disappear in couple of months, so not worth having it.

datapythonista commented on pandas/core/generic.py:
```diff
@@ -1194,44 +1199,54 @@ def rename_axis(self, mapper, axis=0, copy=True, inplace=False):
+        cat            4
+        monkey         2
+        >>> df.index = pd.MultiIndex.from_product(
+        ...             [["mammal"], ['dog', 'cat', 'monkey']])
```
can you try to present this using a better indentation. I think this is not pep8 compatible

datapythonista reviewed (commented):
The example looks great. Added some more comments, minor things that can be improved, and then should be ready to be merged.

bkjchoi72 commented:
@jreback @datapythonista Thanks for the patience guys. Let me make adjustments based on your feedback.

bkjchoi72 committed:
Fixed the indentation and changed the word "alter" to "set" where appropriate.

bkjchoi72 committed:
Merge branch 'master' into GH22895_fix_docstring_for_set_axis_name

datapythonista reviewed (approved):
lgtm, thanks @bkjchoi72 for the fixes

datapythonista commented:
@bkjchoi72 can you take a look at the errors in travis please? they look like genuine errors in your PR.

bkjchoi72 committed:
Fixed failing docstring test

bkjchoi72 committed:
Merge branch 'master' into GH22895_fix_docstring_for_set_axis_name

bkjchoi72 commented:
@datapythonista The tests are fixed. Thanks for your time and guidance!

datapythonista commented:
did you run `flake8 --doctests pandas/core/generic.py`? I think there is a wrong indentation in your example (note that the command will show errors in other parts of the file that you can ignore).

bkjchoi72 commented:
@datapythonista I don't see indentation errors related to my changes. I modified lines 1120 - 1191 and 1207 -1258. Below is the output of `flake8 --doctests pandas/core/generic.py`.
```
$ flake8 --doctests pandas/core/generic.py
pandas/core/generic.py:659:13: F821 undefined name 'p'
pandas/core/generic.py:660:13: F821 undefined name 'p'
pandas/core/generic.py:3195:13: F821 undefined name 'df'
pandas/core/generic.py:3200:13: F821 undefined name 'df'
pandas/core/generic.py:3205:13: F821 undefined name 'df'
pandas/core/generic.py:3211:13: F821 undefined name 'df'
pandas/core/generic.py:3218:13: F821 undefined name 'df'
pandas/core/generic.py:3222:13: F821 undefined name 'df'
pandas/core/generic.py:3227:13: F821 undefined name 'df'
pandas/core/generic.py:7545:71: F721 syntax error in doctest
pandas/core/generic.py:7548:13: F821 undefined name 's'
pandas/core/generic.py:7556:13: F821 undefined name 's'
pandas/core/generic.py:7567:13: F821 undefined name 's'
pandas/core/generic.py:7599:52: F721 syntax error in doctest
pandas/core/generic.py:7603:13: F821 undefined name 'df2'
```
None in the range that I modified shows up from my end. Do you have specific line numbers that are violating pep8 standards?

datapythonista commented on pandas/core/generic.py:
```diff
@@ -1154,29 +1154,40 @@ def rename_axis(self, mapper, axis=0, copy=True, inplace=False):
-        2    3  6
+        >>> df = pd.DataFrame({"num_legs": [4, 4, 2],
+        ...                    "num_arms": [0, 0, 2]},
+        ...                    ["dog", "cat", "monkey"])
```
this is where I think the `['dog'...` should be indented as the `{`, and not one more. Not sure why is not being captured.

datapythonista replied:
doesn't make sense to me, I think it's a flake8 bug, but let's leave it like this is the linting fails with what I think it's the right indentaiton.

bkjchoi72 replied:
Actually, it makes sense to adjust line 1174 only since the argument is composed of two things: dict and list

bkjchoi72 replied:
flake8 still passes with the suggested indentation.

bkjchoi72 committed:
Adjusted indentation

datapythonista commented:
Sorry for the delay @bkjchoi72, can you fix the conflicts, so we can merge?

datapythonista commented:
@bkjchoi72, do you have time to fix the conflicts?

bkjchoi72 commented:
@datapythonista I'll get to it this weekend.

datapythonista committed:
Merging from master and fixing validation issues in the rename_axis docstring

datapythonista commented:
Rebased, and added some more fixes to the docstring.

@Dr-Irv @jreback can you take a look?

Dr-Irv commented:
@datapythonista I was on vacation for the past 10 days, so just saw this now.  All looks good to me.
